hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -222,6 +222,40 @@ func (c *armTypeCreator) convertPropertiesToARMTypes(t *astmodel.ObjectType, isS
 			// all resource Spec Name properties must be strings on their way to ARM
 			// as nested resources will have the owner etc added to the start:
 			result = result.WithProperty(prop.WithType(astmodel.StringType))
+		} else if prop.PropertyType().Equals(astmodel.ResourceReferenceTypeName) || prop.PropertyType().Equals(astmodel.NewOptionalType(astmodel.ResourceReferenceTypeName)) {
+			isRequired := prop.PropertyType().Equals(astmodel.ResourceReferenceTypeName)
+
+			// Extract expected property name
+			values, ok := prop.Tag(astmodel.ARMReferenceTag)
+			if !ok {
+				errs = append(errs, errors.Errorf("ResourceReference property missing %q tag", astmodel.ARMReferenceTag))
+				continue
+			}
+
+			if len(values) != 1 {
+				errs = append(errs, errors.Errorf("ResourceReference %q tag len(values) != 1", astmodel.ARMReferenceTag))
+				continue
+			}
+
+			armPropName := values[0]
+
+			// Remove the property we had since it doesn't apply to ARM
+			result = result.WithoutProperty(prop.PropertyName())
+
+			newProp := astmodel.NewPropertyDefinition(
+				c.idFactory.CreatePropertyName(armPropName, astmodel.Exported),
+				c.idFactory.CreateIdentifier(armPropName, astmodel.NotExported),
+				astmodel.StringType)
+
+			if isRequired {
+				// TODO: This reads very confusingly -- MakeRequired does more than control the Required field though...
+				newProp = newProp.MakeRequired().SetRequired(false)
```

Comment:
Updated comment to explain. I actually think a solution to this problem might be to change the name of this method from `SetRequired` to `WithKubebuilderValidationRequired` and `WithoutKubebuilderValidationRequired`?

Reply:
If folks agree with this, I can send a follow-up PR with this smaller fix?

Reply:
I've actually got this in a branch and think it's an ok idea. I looked into getting rid of this property altogether and just using `ValidatedType` but unfortunately that takes a bit more work as today `ValidatedType` doesn't actually support this validation (it's done as a special case in `PropertyDefinition`, which isn't crazy because this `Required` field really is specific to properties (I don't think it makes sense on a type on its own?)